File: Children-Competition-Java/mpp-proiect-java-ABNegrea/Problema5/src/main/java/Concurs/Service/UserService.java
package Concurs.Service;

import Concurs.Domain.User;
import Concurs.Repository.UserDBRepository;

public class UserService {
    private UserDBRepository userRepo;

    public UserService(UserDBRepository userRepo) {
        this.userRepo = userRepo;
    }

    public User findByEmailPassword(String email, String password) throws Exception {
        User user = userRepo.findByEmailPassword(email,password);
        if(user == null)
            throw new Exception("User inexistent");
        return user;

    }
}

File: Children-Competition-C#/Service/UserService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Concurs.Domain;
using Concurs.Repository;

namespace Concurs.Service
{
    internal class UserService
    {
        private UserDBRepository UserDBRepository;

        public UserService(UserDBRepository UserDBRepository)
        {
            this.UserDBRepository = UserDBRepository;
        }

        public User FindByEmailPassword(string email, string password)
        {
            User user = UserDBRepository.FindByEmailPassword(email, password);
            return user;
        }
    }
}
